lib/cli-engine/file-enumerator.js

```diff
@@ -433,9 +433,10 @@ class FileEnumerator {
         // Enumerate the files of this directory.
         for (const entry of readdirSafeSync(directoryPath)) {
             const filePath = path.join(directoryPath, entry.name);
+            const fileInfo = entry.isSymbolicLink() ? statSafeSync(filePath) : entry;
```

Comment:
`statSafeSync` can return `null` if the link's target had been removed, and then `fileInfo.isFile()` below throws `TypeError: Cannot read property 'isFile' of null`.

I think we should handle that to completely restore v6 behavior. The [v6.8.0 version of `_iterateFilesRecursive`](https://github.com/eslint/eslint/blob/9738f8cc864d769988ccf42bb70f524444df1349/lib/cli-engine/file-enumerator.js#L381) used to always check whether the `stat` object exists, so broken links were ignored instead of throwing error.